alacritty/src/display/window.rs

```diff
@@ -224,6 +224,12 @@ impl Window {
         self.window.set_visible(visibility);
     }
 
+    #[cfg(target_os = "macos")]
+    #[inline]
+    pub fn focus_window(&self) {
+        self.window.focus_window();
```

Comment:
I'd argue that maybe always calling focus on macOS will be fine, but it seems rather strange to me that apps should do that and not macOS itself...

Adding option for that is not a good way though.

Reply:
> A newly created window should automatically receive focus. If it doesn't your window manager is broken.

> but it seems rather strange to me that apps should do that and not macOS itself...

This is the default window manager for macOS 15.1. I don't think it's broken or strange, just different.

When [`window.set_visible(true)`](https://github.com/alacritty/alacritty/blob/d552c6b251d199b4913754dd25fca330f87651fd/alacritty/src/display/mod.rs#L485) is called, winit's [AppKit delegate](https://github.com/rust-windowing/winit/blob/d3207a8d76c7b6a6283dee36c19c1dac4e378fa7/src/platform_impl/apple/appkit/window_delegate.rs#L921) calls [`NSWindow.makeKeyAndOrderFront`](https://developer.apple.com/documentation/appkit/nswindow/1419208-makekeyandorderfront) which moves the window to the front of the application's screen list and makes it the key window for that application. If the application itself is already active, then great, the window has focus. If the application isn't active, as in my use-case, then the window doesn't get focus.

Not stealing focus from other apps seems sensible to me, but regardless, that's how the macOS window manager seems to work.

> I'd argue that maybe always calling focus on macOS will be fine,

> Adding option for that is not a good way though.

Fair enough. I set it an option because I didn't know how you want this to behave and I generally don't like applications stealing focus unless asked 😄

Would you be willing to reconsider if I change to always call focus for macOS? That's a far less intrusive change.

Reply:
> Would you be willing to reconsider if I change to always call focus for macOS? That's a far less intrusive change.

That's how it behaves on other platforms so I don't think I'd mind it too much.

Reply:
Can you please reopen this PR? Otherwise I can create a new one.

Reply:
I think it's fine to just create a new one. Seems significant enough difference.